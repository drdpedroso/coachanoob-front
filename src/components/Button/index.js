import React from 'react'
import PropTypes from 'prop-types';
import {Button} from 'semantic-ui-react'

const BaseButton = (props) => {
    return (
        <Button
            {...props}
            onClick={props.onClick}
        >
            {props.children}
        </Button>
    )
};

BaseButton.propTypes = {
    onClick: PropTypes.func,
    children: PropTypes.node
};

export default BaseButton
