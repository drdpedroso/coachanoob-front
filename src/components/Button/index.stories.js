import React from 'react';
import {storiesOf} from '@storybook/react';
import {action} from '@storybook/addon-actions';

import Button from './';
import {Icon} from "semantic-ui-react";

storiesOf('Button', module)
    .add('with text primary', () => (
        <Button primary onClick={action('clicked')}>Hello Button</Button>
    ))
    .add('with text secondary', () => (
        <Button secondary onClick={action('clicked')}>Hello Button</Button>
    ))
    .add('with icon primary', () => (
        <Button primary onClick={action('clicked')}><Icon name='world'/></Button>
    ))
    .add('with icon secondary', () => (
        <Button secondary onClick={action('clicked')}><Icon name='world'/></Button>
    ))
    .add('with texxt primary fixed bottom', () => (
        <Button attached='bottom' onClick={action('clicked')}>Hello Button</Button>
    ))
    .add('with text primary fluid', () => (
        <Button fluid onClick={action('clicked')}>Hello Button</Button>
    ));
