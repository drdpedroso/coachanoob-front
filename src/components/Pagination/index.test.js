import React from 'react';
import Button from './index'
import renderer from 'react-test-renderer';


describe('Pagination', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<Button>Menor Bolado</Button>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
