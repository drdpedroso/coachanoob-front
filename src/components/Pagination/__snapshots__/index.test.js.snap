// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders correctly 1`] = `
<div
  aria-label="Pagination Navigation"
  className="ui pagination menu"
  role="navigation"
>
  <a
    aria-current={false}
    aria-disabled={false}
    aria-label="First item"
    className="item"
    onClick={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
    type="firstItem"
    value={1}
  >
    «
  </a>
  <a
    aria-current={false}
    aria-disabled={false}
    aria-label="Previous item"
    className="item"
    onClick={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
    type="prevItem"
    value={NaN}
  >
    ⟨
  </a>
  <a
    aria-current={false}
    aria-disabled={false}
    className="item"
    onClick={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
    type="pageItem"
    value={1}
  >
    1
  </a>
  <a
    aria-current={false}
    aria-disabled={true}
    className="item"
    onClick={[Function]}
    onKeyDown={[Function]}
    tabIndex={-1}
    type="ellipsisItem"
    value={NaN}
  >
    ...
  </a>
  <a
    aria-current={false}
    aria-disabled={true}
    className="item"
    onClick={[Function]}
    onKeyDown={[Function]}
    tabIndex={-1}
    type="ellipsisItem"
    value={NaN}
  >
    ...
  </a>
  <a
    aria-current={false}
    aria-disabled={false}
    aria-label="Next item"
    className="item"
    onClick={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
    type="nextItem"
    value={NaN}
  >
    ⟩
  </a>
  <a
    aria-current={false}
    aria-disabled={false}
    aria-label="Last item"
    className="item"
    onClick={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
    type="lastItem"
    value={NaN}
  >
    »
  </a>
</div>
`;
